nautobot_golden_config/nornir_plays/config_deployment.py

```diff
@@ -28,31 +30,36 @@ def run_deployment(task: Task, logger: NornirLogger, commit: bool, config_plan_q
 
     if commit:
         plans_to_deploy.update(status=Status.objects.get(slug="in-progress"))
-        result = task.run(
-            task=dispatcher,
-            name="DEPLOY CONFIG TO DEVICE",
-            method="merge_config",
-            obj=obj,
-            logger=logger,
-            config=consolidated_config_set,
-            default_drivers_mapping=get_dispatcher(),
-        )[1].result["result"]
-        if not result:
-            plans_to_deploy.update(status=Status.objects.get(slug="failed"))
-            logger.log_failure(obj=obj, message="No Nornir Result was Returned.")
-        else:
-            if not result.failed:
-                logger.log_success(obj=obj, message="Successfully deployed configuration to device.")
-                plans_to_deploy.update(status=Status.objects.get(slug="completed"))
-            else:
+        try:
+            result = task.run(
+                task=dispatcher,
+                name="DEPLOY CONFIG TO DEVICE",
+                method="merge_config",
+                obj=obj,
+                logger=logger,
+                config=consolidated_config_set,
+                default_drivers_mapping=get_dispatcher(),
+            )[1]
+            task_changed, task_result, task_failed = result.changed, result.result, result.failed
+            if task_changed and task_failed:
+                # means config_revert happened in `napalm_configure`
                 plans_to_deploy.update(status=Status.objects.get(slug="failed"))
                 logger.log_failure(obj=obj, message="Failed deployment to the device.")
+            elif not task_changed and not task_failed:
+                plans_to_deploy.update(status=Status.objects.get(slug="Completed"))
+                logger.log_failure(obj=obj, message="Nothing was deployment to the device.")
+            else:
+                if not task_failed:
+                    logger.log_success(obj=obj, message="Successfully deployed configuration to device.")
+                    plans_to_deploy.update(status=Status.objects.get(slug="completed"))
+            return Result(host=task.host, result=task_result)
+        except NornirSubTaskError:
+            plans_to_deploy.update(status=Status.objects.get(slug="failed"))
+            logger.log_failure(obj=obj, message="Failed deployment to the device.")
```

Comment:
We don't need to return a failed result here?

Reply:
doesn't seem its needed.  Did additinoal testing and this is working as designed.  Most likely because the original subtask run is actually running through the processor.py that handles identifying failures.

Reply:
Either way I'll plan on returning a result here to be complete.  commit to come